okhttp/src/test/java/okhttp3/internal/tls/CertificatePinnerChainValidationTest.java

```diff
@@ -102,23 +103,12 @@ public void setup(MockWebServer server) {
 
     // The request should complete successfully.
     server.enqueue(new MockResponse()
-        .setBody("abc")
-        .setSocketPolicy(SocketPolicy.DISCONNECT_AT_END));
+        .setBody("abc"));
     Call call1 = client.newCall(new Request.Builder()
         .url(server.url("/"))
         .build());
     Response response1 = call1.execute();
     assertThat(response1.body().string()).isEqualTo("abc");
-
-    // Confirm that a second request also succeeds. This should detect caching problems.
-    server.enqueue(new MockResponse()
```

Comment:
@swankjesse removing some code you added years ago.  I'm not confident about this fix, except that it seems really flaky...